tests/test_processors.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+from lxml.html import fromstring
+from parsel import Selector, SelectorList
+from web_poet import HttpResponse, field
+from zyte_parsers import Breadcrumb as zp_Breadcrumb
+
+from zyte_common_items import BasePage, Breadcrumb, ProductPage
+from zyte_common_items.processors import breadcrumbs_processor
+
+breadcrumbs_html = """
+<div class="pagesbar">
+  <ul>
+    <li><a href="/blog/">Home</a></li>
+    <li><a href="/blog/about/">About</a></li>
+  </ul>
+</div>
+"""
+
+breadcrumbs_expected = [
+    Breadcrumb(name="Home", url="http://www.example.com/blog/"),
+    Breadcrumb(name="About", url="http://www.example.com/blog/about/"),
+]
+
+
+@pytest.mark.parametrize(
+    "input_value,expected_value",
+    [
+        (None, None),
+        ([], []),
+        ("foo", "foo"),
+        (Selector(text="<html></html>"), None),
+        (SelectorList([]), None),
+        (breadcrumbs_expected, breadcrumbs_expected),
+        (fromstring(breadcrumbs_html), breadcrumbs_expected),
+        (Selector(text=breadcrumbs_html), breadcrumbs_expected),
+        (
+            [
+                zp_Breadcrumb(name="Home", url="http://www.example.com/blog/"),
+                zp_Breadcrumb(name="About", url="http://www.example.com/blog/about/"),
```

Comment:
To ensure we cover the use case, could you please add a test for a page object which calls zyte_parsers.extract_breadcrumbs itself, and returns the result?

Reply:
We actually don't :)